Assets/Scripts/Strategy/GameResources/IResource.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SwordAndBored.Strategy.GameResources
+{
+    public interface IResource
```

Comment:
Make this extend the `Utilities.IObserver<T>` interface. Probably of type `IResourceSubscriber`